inc/endpoints/class-data-endpoint.php

```diff
@@ -46,8 +46,9 @@ public function register_rest_routes() {
 				self::get_namespace(),
 				'/data/' . $endpoint['slug'],
 				[
-					'methods'  => \WP_REST_Server::READABLE,
-					'callback' => $endpoint['callback'],
+					'callback'            => $endpoint['callback'],
+					'methods'             => \WP_REST_Server::READABLE,
+					'permission_callback' => '__return_true',
```

Comment:
Not sure how data endpoints are being used by integrators, but I wonder if we should use a similar pattern to how callbacks are registered and set the `permission_callback` to `$endpoint['permission_callback']` so integrations can supply their own permissions callback.